laborator/content/operatii-memorie-gdb/solution/1-iterate/iterate.c

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+
+int main(void)
+{
+	unsigned int i;
+	int v[] = {0xCAFEBABE, 0xDEADBEEF, 0x0B00B135, 0xBAADF00D, 0xDEADC0DE};
+	/*
+	 * vom folosi pointeri către elemente de dimensiuni diferite pentru a
+	 * ușura afișarea elementelor de 1, 2 și 4 bytes
+	 */
+	unsigned char *char_ptr = (unsigned char *) &v;
+	unsigned short *short_ptr = (unsigned short *) &v;
+	unsigned int *int_ptr = (unsigned int *) &v;
+
+	// loop pentru fiecare byte al vectorului v
+	for (i = 0 ; i < sizeof(v); ++i) {
+		printf("%p -> 0x%x\n", char_ptr, *char_ptr);
+		++char_ptr;
+	}
+	printf("-------------------------------\n");
+
+	/*
+	 * loop din 2 în 2 bytes, avem doar jumătate din pași fiindcă afișăm câte
+	 * 2 bytes la fiecare pas
+	 */
+	for (i = 0 ; i < sizeof(v) / 2; ++i) {
```

Comment:
```suggestion
	for (i = 0 ; i < sizeof(v) / sizeof(*short_ptr); ++i) {
```

Reply:
> there's a makefile setting turned on
You mean a `gcc` flag: `-Werror`. But I don't get this warning on my machine. Which `gcc` version are you using? Mine is `gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1)`.

Either way, to disable this warning, try redefining `CFLAGS` [1] in `solution/1-iterate/Makefile` like so: `CFLAGS := -m32 -g -Wall -Werror -fno-pic -masm=intel` **before including the generic makefile**.